### DISCUSSION TITLE
Is there a way to "uneval" Zod Schema - Convert schema to stringSchema

### DISCUSSION THREAD

**Discussion by timheerwagen** [colinhacks/zod#2969] [Q&A]
Hello,

I have a Zod scheme, for example

`const schema = z.string()`

Now I want to convert the schema into a string that represents this schema so that I can save it later in a file:

`const schemaString = "z.string()"`

I have tried everything, e.g. packages like `lave`, `devalue`, `tosource`, `serialize-javascript`, `jsesc`, `superjson`....

My only working solution at the moment is to convert to json and then back to string with `zod-to-json-schema` and `json-schema-to-zod`. 

```ts

import zodToJsonSchema from "zod-to-json-schema";
import { parseSchema } from "json-schema-to-zod";

 parseSchema(
    zodToJsonSchema(schema, {
      $refStrategy: "none",
    })
  )
    .split(".catchall(z.never())")
    .join(".strict()")
```

But I don't like that because the input is not always equal to the output, for example coerce doesn't work. As you can see, I already had some workarounds to implement. The code does not have to run in production, but only in the dev environment.

I would also appreciate an easier way to copy the schema into a new file.

**Comment by sohanemon** [comment]
I don't know why you want to achieve this. As I know you can't convert from `z.string()` to `"z.string()"`.

But for my use case I found a solution.

```tsx
<FormLabel>{convertToNormalCase(name)}<span className="inline-block pl-0.5 text-red-500">{!JSON.stringify(schema).includes('ZodOptional') && '*'} </span> </FormLabel>
```
                                
thank you

**Reply by timheerwagen** [reply]
The only reason I want to achieve this is because I want to save a Zod schema as a file and I use ts-morph to do this.

**Reply by Serpentarius13** [reply]
In my case I want to create zod schema definitions for my SvelteKit project so I can use drizzle's zod parser in my trpc handlers and on client both

**Reply by ilyichv** [reply]
hey @timheerwagen did you manage to find a solution?

**Reply by timheerwagen** [reply]
Yes, I have completely changed the way my codegen works and do not generate the Zod schema directly. Instead, I have created an object with all my specific schemas and generate the imports via codegen.
